api/provider/aws/service_delete.go

```diff
@@ -1,5 +1,130 @@
 package aws
 
-func (s *ServiceProvider) Delete(ServiceID string) error {
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
+)
+
+// Delete a specified service within a cluster.
+func (s *ServiceProvider) Delete(serviceID string) error {
+	desiredCount := 0
+
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		if err, ok := err.(*errors.ServerError); ok && err.Code == errors.ServiceDoesNotExist {
+			return nil
+		}
+
+		return err
+	}
+
+	clusterName := addLayer0Prefix(s.Config.Instance(), environmentID)
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+
+	service, err := s.readService(clusterName, fqServiceID)
+	if err != nil {
+		return err
+	}
+
+	taskARNs, err := s.getServieTasks(clusterName, service.Deployments)
```

Comment:
typo

Reply:
should't we change this to `getServiceRunningTasks` because we don't need to stop tasks that are already stopped?